sktime/transformations/series/bootstrapping.py

```diff
@@ -0,0 +1,401 @@
+# -*- coding: utf-8 -*-
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Bootstrapping methods for time series."""
+
+__author__ = ["ltsaprounis"]
+
+from copy import copy
+from typing import Tuple
+
+import numpy as np
+import pandas as pd
+from statsmodels.tsa.api import STL as _STL
+
+from sktime.transformations.base import BaseTransformer
+from sktime.transformations.series.boxcox import BoxCoxTransformer
+
+
+class BootsrappingTransformer(BaseTransformer):
+    """Creates a population of similar time series.
+
+    This method utilises a form of bootstrapping to generate a population of
+    similar time series to the input time series [1]_, [2]_.
```

Comment:
Hyndman (I assume section 12.5?) does not mention the BoxCox transformer explicitly.
Personally, I would interpret the text that the bootstrap is STL plus the mentioned block bootstrap method.

Reply:
Yes it's a bit vague in the 3rd edition, but the original reference is Bergmeir 2016. In the second edition of forecasting principles and practice, Hyndman mentions BoxCox explicitly ([fpp 2nd edition chapter 11.4](https://otexts.com/fpp2/bootstrap.html)). In general all the implementations and papers I've seen that utilise this bootstrapping method always have BoxCox (with Guerrero to find lambda) as the first step. 

Conceptually this makes sense. The variance needs to be stable for MBB on the residuals to work. The transformed series has stable variance and when the inverse transform is applied in the end, the synthetic series has the same variance pattern as the original series.

Reply:
well, you could also say: Bergmeir is a pipeline [boxcox][bootstrap][bagging], and this estimator should be [bootstrap]
Or, you could say, Bergmeir is [boxcox+bootstrap][bagging], and this estimator should be [boxcox+bootstrap].

If there is no computational advantage to put boxcox and the bootstrap together, I´d personally split them up, and then use a pipeline constrution when implementing the full Bergmeir.

But I have no strong opinion on this. Doing the "combined" version just makes the docstrings more complicated, and may result in us having to implement the pure [bootstrap] separately later (time series bootstrap resampling is generally a common component).